sdk/python/kfp/client/client.py

```diff
@@ -1057,7 +1057,10 @@ def create_run_from_pipeline_func(
             ``RunPipelineResult`` object containing information about the pipeline run.
         """
         #TODO: Check arguments against the pipeline function
-        pipeline_name = pipeline_func.name
+        if isinstance(pipeline_func, base_component.BaseComponent):
+            pipeline_name = pipeline_func.name
+        else:
+            pipeline_name = pipeline_func.__name__
```

Comment:
This one is a bit odd. The [type hint](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L1026) specifies a union:
> pipeline_func: Union[Callable[..., Any], base_component.BaseComponent],

`BaseComponent` has a `name` attribute. A generic callable doesn't. Which is why mypy complained about this. Not sure this is the optimal solution and totally open to alternatives.

Reply:
In this case, I believe the annotation should be `base_component.BaseComponent` only.

The `Callable` annotation is a relic from v1, where pipelines were pure functions. In v2, pipelines created with the `@dsl.pipeline` decorator are instances of `GraphComponent`, which is a subclass of `BaseComponent`.

Reply:
In this case, I believe the annotation should be `base_component.BaseComponent` only.

The `Callable` annotation is a relic from v1, where pipelines were pure functions. In v2, pipelines created with the `@dsl.pipeline` decorator are instances of `GraphComponent`, which is a subclass of `BaseComponent`.